### PR TITLE
Upgrade Django to 4.2 LTS

### LINKED ISSUE #897: Django 4.2 Upgrade
## Description
Under the effort of [Django 4.2 Upgrade](https://github.com/openedx/public-engineering/issues/199), complete all of the following steps to complete the upgrade. 

- [x] Update tox & Github action workflow using modernisers to add support for `Django 4.2`
- [x] Remove any versions of Python earlier than 3.8 from `tox.ini`, and `GitHub Actions` workflows.
- [x] Update the pinned `Django` version in the requirements to `Django==4.2`
- [x] Run `make upgrade` to update all dependencies for `Django 4.2`.
- [x] Run and verify all tests are passing in the CI for both `Django 4.2` and `Django 3.2` to contain backward compatibility.
- [x] Run available code-mods to fix the failing tests. 
- [ ] Add conditional checks wherever needed to still have support for `Django 3.2`. 
- [ ] Update the repo support field in the [IDA Upgrade Sheet](https://openedx.atlassian.net/wiki/spaces/AC/pages/3815637011/Django+4.2+Upgrade+List+of+IDAs).

### PR DESCRIPTION
**Pull request by zubairshakoorarbisoft**
Upgrading  to django42 and dropping django32 support.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -4,43 +4,44 @@
...
 cffi==1.15.1
     # via pynacl
-charset-normalizer==3.1.0
+charset-normalizer==3.2.0
     # via requests
-click==8.1.3
+click==8.1.7
     # via edx-django-utils
-django==3.2.20
+django==3.2.21
```
Comment (awais786) [inline requirements.txt:25]: Version remaind 3.2

File: Makefile [hunk 2]
```diff
@@ -44,7 +44,7 @@ $(COMMON_CONSTRAINTS_TXT):
 
 export CUSTOM_COMPILE_COMMAND = make upgrade
 upgrade: $(COMMON_CONSTRAINTS_TXT) ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
-	sed 's/Django<2.3//g' requirements/common_constraints.txt > requirements/common_constraints.tmp
+	sed 's/Django<4.0/Django<4.3/g' requirements/common_constraints.txt > requirements/common_constraints.tmp
```
Comment (awais786) [inline Makefile:47]: You don't need this.

File: Makefile [hunk 3]
```diff
@@ -44,8 +44,8 @@ $(COMMON_CONSTRAINTS_TXT):
 
 export CUSTOM_COMPILE_COMMAND = make upgrade
 upgrade: $(COMMON_CONSTRAINTS_TXT) ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
-	sed 's/Django<2.3//g' requirements/common_constraints.txt > requirements/common_constraints.tmp
 	mv requirements/common_constraints.tmp requirements/common_constraints.txt
+	sed 's/Django<4.0//g' requirements/common_constraints.txt > requirements/common_constraints.tmp
```
Comment (iamsobanjaved) [inline Makefile:48]:
```suggestion
	sed 's/Django<4.3//g' requirements/common_constraints.txt > requirements/common_constraints.tmp
```

File: requirements.txt [hunk 4]
```diff
@@ -4,43 +4,44 @@
...
 cffi==1.15.1
     # via pynacl
-charset-normalizer==3.1.0
+charset-normalizer==3.2.0
     # via requests
-click==8.1.3
+click==8.1.7
     # via edx-django-utils
-django==3.2.20
+django==4.2.5
```
Comment (awais786) [inline requirements.txt:27]: Version upgraded.

File: xqueue/settings.py [hunk 5]
```diff
@@ -90,8 +95,6 @@
 # Make this unique, and don't share it with anybody.
 SECRET_KEY = 'uofqkujp@z#_vtwct+v716z+^3hijelj1^fkydwo2^pbkxghfq'
 
-DEFAULT_HASHING_ALGORITHM = "sha1"
```
Comment (awais786) [inline xqueue/settings.py:93]: Its removed in django42. Now by default it is 256.

**Review by iamsobanjaved — APPROVED** [review]

**Pull request merged**